packages/mobile/src/send/saga.ts

```diff
@@ -169,12 +188,52 @@ function* sendPaymentOrInviteSaga({
   }
 }
 
+export function* validateRecipientAddressSaga({
+  userInputOfFullAddressOrLastFourDigits,
+  fullAddressValidationRequired,
+  recipient,
+}: ValidateRecipientAddressAction) {
+  Logger.debug(TAG, 'Starting Recipient Address Validation')
+  try {
+    if (!recipient.e164PhoneNumber) {
+      throw Error(`Invalid recipient type for Secure Send: ${recipient.kind}`)
+    }
+
+    const userAddress = yield select(userAddressSelector)
+    const e164NumberToAddress = yield select(e164NumberToAddressSelector)
+    const { e164PhoneNumber } = recipient
+    const possibleRecievingAddresses = e164NumberToAddress[e164PhoneNumber]
```

Comment:
Use the `getAdressFrom....` helpers where you can to help cover edge cases

Reply:
This is an example where using `getAddressFromPhoneNumber` would break it because there are multiple addresses mapped to a phone number but before we've asked the user to validate the "correct" one.

Reply:
I see, agreed :)